systemservices/ethwallet/ethwallet/create.go

```diff
@@ -1,30 +1,22 @@
 package ethwallet
 
 import (
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/mesg-foundation/engine/systemservices/ethwallet/client"
+	"github.com/mesg-foundation/engine/protobuf/types"
 )
 
-type createInputs struct {
-	Passphrase string `json:"passphrase"`
-}
-
-type createOutputSuccess struct {
-	Address common.Address `json:"address"`
-}
-
-func (s *Ethwallet) create(input map[string]interface{}) (map[string]interface{}, error) {
-	var inputs createInputs
-	if err := client.Unmarshal(input, &inputs); err != nil {
-		return nil, err
-	}
-
-	account, err := s.keystore.NewAccount(inputs.Passphrase)
+func (s *Ethwallet) create(inputs *types.Struct) (*types.Struct, error) {
+	account, err := s.keystore.NewAccount(inputs.Fields["passphrase"].GetStringValue())
 	if err != nil {
 		return nil, err
 	}
 
-	return client.Marshal(createOutputSuccess{
-		Address: account.Address,
-	})
+	return &types.Struct{
```

Comment:
Isn't possible to keep the easy to use `createInputs` and `createOutput` struct and use convert marshal/unmarshal function to convert from those struct to `*types.Struct`?

Reply:
The problem here is api uses `*types.Struct` as inputs, outputs and data. So there are two option 

1) keeps `map[string]interface{}` (or any other type) and converts it to `*types.Struct`
2) Use `*types.Struct` directly

We wanted to get rid of 1 solution, so we go with 3. I can create helper functions for getting fields values or createing values, but that's all.

3) any proposition???

Reply:
just for reference: https://godoc.org/github.com/robteix/protoconv